Dockerfile

```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile:1
 
-FROM golang:1.17
+FROM golang:1.21
```

Comment:
Let's use the same version of go in the Dockerfile as the one in go.mod.
This ensures that our development and deployment environments are the same.

Reply:
I have updated the version to 1.21 at both places, Dockerfile as well as go.mod and updated all the packages as well.